lexicon/providers/safedns.py

```diff
@@ -0,0 +1,206 @@
+            if 'meta' in payload \
+                    and 'pagination' in payload['meta'] \
+                    and 'links' in payload['meta']['pagination'] \
+                    and 'next' in payload['meta']['pagination']['links']:
+                next_url = payload['meta']['pagination']['links']['next']
+            else:
+                next_url = None
+
+            # Assign the returned attributes to the keys that lexicon expects
+            for record in payload['data']:
+                record = self._clean_TXT_record(record)
+                processed_record = {
+                    'id': record['id'],
+                    'name': record['name'],
+                    'type': record['type'],
+                    'content': record['content'],
+                    'updated_at': record['updated_at'],
+                    'ttl': record['ttl'],
+                    'priority': record['priority']
+                }
+                records.append(processed_record)
+
+        # This is filtering logic to return only the record which matches what has
+        # been passed in to the method
+        if rtype:
+            records = [
+                record for record in records
+                if record['type'] == rtype]
+        if name:
+            records = [
+                record for record in records
+                # DNS should not be case-sensitive, so perform lower-case comparison
+                if record['name'].lower() == self._full_name(name).lower()]
+        if content:
+            records = [
+                record for record in records
+                if record['content'].lower() == content.lower()]
+
+        LOGGER.debug('list_records: %s', records)
+        return records
+
+    def _create_record(self, rtype, name, content):
+
+        # Check whether the record already exists with the same  rtype, name & content.
+        # If so, claim to have added the record, but dont't do anything.
+        records = self._list_records(rtype, name, content)
+        if records:
+            LOGGER.debug('create_record: (ignored, duplicate record): %s', records[0]['id'])
+            return True
+
+        # Make sure TXT records are wrapped in quotes
+        if content:
+            content = self._add_quotes(rtype, content)
+
+        data = {
+            'name': self._full_name(name),
+            'type': rtype,
+            'content': content
+        }
+
+        self._post('/zones/{0}/records'.format(self.domain), data)
+        LOGGER.debug('create_record: %s', True)
+        return True
+
+    def _update_record(self, identifier, rtype=None, name=None, content=None):
+
+        # Make sure the update won't cause a duplicate entry. If it will, fail silently
+        records = self._list_records(rtype, name, content)
+        if len(records) >= 1:
+            LOGGER.debug('update_record: (ignored, update would cause a duplicate record): %s',
+                         records[0]['id'])
+            return True
```

Comment:
Just one last thing, I think you can remove this logic now.

Reply:
Yes, annoyed I missed that. Corrected and pushed